wgpu/gui/wx.py

```diff
@@ -28,37 +29,48 @@ def __init__(self, callback):
         self._callback = callback
 
     def Notify(self, *args):  # noqa: N802
-        self._callback()
+        try:
+            self._callback()
+        except RuntimeError:
+            pass  # wrapped C/C++ object of type WxWgpuWindow has been deleted
 
 
 class WxWgpuWindow(WgpuCanvasBase, wx.Window):
     """A wx Window representing a wgpu canvas that can be embedded in a wx application."""
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, *args, max_fps=30, **kwargs):
         super().__init__(*args, **kwargs)
 
+        # Variables to limit the fps
+        self._draw_time = 0
+        self._max_fps = float(max_fps)
+        self._request_draw_timer = TimerWithCallback(self.Refresh)
+
+        # We also keep a timer to prevent draws during a resize. This prevents
+        # issues with mismatching present sizes during resizing (on Linux).
+        self._resize_timer = TimerWithCallback(self._on_resize_done)
         self._draw_lock = False
-        self._resize_timer = TimerWithCallback(self._request_draw)
 
         self.Bind(wx.EVT_PAINT, self.on_paint)
         self.Bind(wx.EVT_ERASE_BACKGROUND, lambda x: None)
-        self.Bind(wx.EVT_SIZE, lambda e: self._request_draw_later())
+        self.Bind(wx.EVT_SIZE, self._on_resize)
 
     def on_paint(self, event):
-        if self._draw_lock:
-            return
+        self._draw_time = time.perf_counter()
         dc = wx.PaintDC(self)  # needed for wx
-        self._draw_frame_and_present()
+        if not self._draw_lock:
+            self._draw_frame_and_present()
         del dc
```

Comment:
should this by in a try/finally maybe?

Reply:
The `_draw_frame_and_present` already does a try-except on the actual draw function:
https://github.com/pygfx/wgpu-py/blob/e838db358e68a95771fd1f6ce186b335c6722bf3/wgpu/gui/base.py#L99-L114